telemetry/core/src/main/scala/io/buoyant/telemetry/MetricsTree.scala

```diff
@@ -12,6 +12,19 @@ trait MetricsTree {
 
   def metric: Metric
 
+  def counter: Option[Metric.Counter] = metric match {
+    case c: Metric.Counter => Some(c)
+    case _ => None
+  }
+  def gauge: Option[Metric.Gauge] = metric match {
+    case g: Metric.Gauge => Some(g)
+    case _ => None
+  }
+  def stat: Option[Metric.Stat] = metric match {
+    case s: Metric.Stat => Some(s)
+    case _ => None
+  }
+
```

Comment:
well, the PF should def be val'd... i meant the option --  usually this is just an intermediary allocation like

`t.metric match {
  case c: Metric.Counter =>
    // do the thing
  case _ =>
    // don't do it
}
```

By providing `counter`, we encourage callers to do a needless option-allocation.  it's not a huge deal, i just tend to prefer public APIs that are minimal and don't encourage undesirable patterns